### PR TITLE
add usage tracking for clicking the `ChangedFilesCount` component

### PR DESCRIPTION
**Pull request by annthurium**
### Description of the Change

In https://github.com/atom/github/issues/1573, we discussed discoverability of the Atom package.  @simurai proposed changing the status bar tile to say "Git" instead of "n file(s)".  I think that's a really good idea, but before implementing that change, I'd like to instrument this component and establish a baseline.  That way we'll know if there's a correlation between changing the text and how often it's clicked, even if causation is harder to establish. :-p 

This test also adds unit tests for `ChangedFilesCountView` since there previously were none. 

### Alternate Designs

Based on conversations with @telliott27, I'm using the `addEvent` api instead of the `incrementCounter` api.  In the past, client apps has used counters for feature usage.  However, counters don't retain metadata for each individual event, so `addEvent` gives us a fuller picture of what's going on.

### Benefits

The ability to understand how users are interacting with this component.  More broadly, I want to understand if users are having trouble discovering the github package.

### Possible Drawbacks

None that I can think of. 

### Applicable Issues

https://github.com/atom/github/issues/1573

### CODE REVIEW

File: lib/views/changed-files-count-view.js [hunk 1]
```diff
@@ -24,7 +30,7 @@ export default class ChangedFilesCountView extends React.Component {
       <button
         ref="changedFiles"
         className="github-ChangedFilesCount inline-block"
-        onClick={this.props.didClick}>
+        onClick={this.handleClick.bind(this)}>
```
Comment (annthurium) [inline lib/views/changed-files-count-view.js]: @smashwilson : would it be better to use `autobind` here for consistency's sake even though there's only one method to bind and no constructor in this component?

Comment (annthurium) [reply lib/views/changed-files-count-view.js]: @smashwilson thanks for the extra context!  I was just being lazy. :-p . I'll fix it.

**Comment by smashwilson** [comment]
> This pull request also adds unit tests for ChangedFilesCountView since there previously were none.

Yesss 🥇 

So somewhat awkwardly, a bunch of the tests for the status bar components are smashed up into [`test/controllers/status-bar-tile-controller.test.js`](https://github.com/atom/github/blob/dfcfc645cabfcf59ff5741b7b7fa63f413aa9f4a/test/controllers/status-bar-tile-controller.test.js). There _are_ a few for the change files view:

https://github.com/atom/github/blob/dfcfc645cabfcf59ff5741b7b7fa63f413aa9f4a/test/controllers/status-bar-tile-controller.test.js#L649-L680

But they're pretty sparse and quite old. I'd say delete those in favor of yours ✂️

**Comment by annthurium** [comment]
@smashwilson some of the tests in `status-bar-tile-controller.test.js` are exercising functionality that couldn't be tested by the `ChangedFilesCountView` in isolation, without breaking encapsulation.  For example, testing that clicking the `ChangedFilesCountView` actually toggles the tab, or testing that the status tile icons don't show up if the project lacks a repository.

There is one test in `status-bar-tile-controller` that does feels redundant tho, so I'll remove that one.
